tests/densify/test_mvs_patchmatchnet.py

```diff
@@ -123,6 +129,48 @@ def test_patchmatchnet_result(self) -> None:
 
         self.assertTrue(mean_k_yz_error < 0.2 and x_error < 0.5)
 
+    def test_compute_filtered_reprojection_error(self) -> None:
+        """Test whether the compute_filtered_reprojection_error produces correct reprojection errors
+        In this test, we assume there is an object O at (0, 0, 0), where all cameras look at.
+        Then we select camera 0 as dummy reference view, while camera 1 as dummy source view.
+        """
+        # prepare PatchmatchNet dataset
+        dataset = PatchmatchNetData(images=self._img_dict, sfm_result=self._sfm_result, max_num_views=NUM_IMAGES)
+
+        # set dummy reference and source view pair
+        dummy_ref_view = 0
+        dummy_src_view = 1
+
+        # fetch depthmap resolution
+        height, width = self._img_dict[dummy_ref_view].value_array.shape[:2]
+
+        # set dummy estimated depthmap for reference and source view
+        dummy_ref_depthmap = np.zeros([1, height, width])
+        dummy_ref_depthmap[0, height // 2, width // 2] = np.linalg.norm([CAMERA_CIRCLE_RADIUS, CAMERA_HEIGHT])
+
+        dummy_src_depthmap = np.zeros([1, height, width])
+        # add some error when estimating the depthmap of source view
+        dummy_src_depthmap[0, height // 2, width // 2] = np.linalg.norm([CAMERA_CIRCLE_RADIUS, CAMERA_HEIGHT]) + 0.5
+
+        # build depthmap list
+        depth_list = {dummy_ref_view: dummy_ref_depthmap, dummy_src_view: dummy_src_depthmap}
+
+        # set dummy joint mask to only include the image center, where object O is located
+        dummy_joint_mask = np.zeros([height, width], dtype=np.bool)
+        dummy_joint_mask[height // 2, width // 2] = True
+
+        # calculate reprojection error
+        reproject_errors = compute_filtered_reprojection_error(
+            dataset=dataset,
+            ref_view=dummy_ref_view,
+            src_views=[dummy_src_view],
+            depth_list=depth_list,
+            max_reprojection_err=MAX_GEOMETRIC_PIXEL_THRESH,
+            joint_mask=dummy_joint_mask,
+        )
+
+        self.assertAlmostEqual(reproject_errors[0], 0.828, 2)
```

Comment:
@codyly could you explain where the 0.828 number comes from? a simple derivation / math for it? what is the 3d point and what is the uv coordinate?

Reply:
can we also ensure that the rest of the image has its reproj errors filtered out? should the list be of length 1?